### ISSUE TITLE
[Bug]: 

### ISSUE THREAD

**Issue by asplamagnifique2** [mattermost/desktop#2768]
### Before you file a bug report

- [X] I have checked the [issue tracker](https://github.com/mattermost/desktop/issues) and have not found an issue that matches the one I'm filing.
- [X] This issue is not a troubleshooting question. Troubleshooting questions go here: https://forum.mattermost.com/c/trouble-shoot/16.
- [X] This issue is not a feature request. You can request features and make product suggestions here: https://mattermost.com/suggestions/.
- [X] This issue doesn't reproduce on web browsers (such as in Chrome). If it does, [issue reports go to the Mattermost Server repository](https://github.com/mattermost/mattermost-server/issues).
- [X] This issue reproduces on the most recent [stable version](https://github.com/mattermost/desktop/releases/latest), or the most recent [prerelease version](https://github.com/mattermost/desktop/releases) of the Mattermost Desktop App.
- [X] I have read the [contribution guidelines](https://github.com/mattermost/desktop/blob/master/CONTRIBUTING.md).

### Mattermost Desktop Version

5.4

### Operating System

mac OS 13.4.1 Macbook Air M1

### Mattermost Server Version

7.10.0

### Steps to reproduce

I had a mattermost app in 5.3.1.
I've updated to 5.4 this morning.

### Expected behavior

I want to switch to channel / PM on the app.

### Observed behavior

Since, I'm not able to switch channel.
An error message is display. I can't go back to the main channel I have to CTRL+F+5 or close the app and reopen.

### Log Output

```shell
[2023-06-22 11:57:46.790] [debug] [DownloadsManager] init { 'this.json': {}, validatedJSON: {} }
[2023-06-22 11:57:46.793] [debug] [DownloadsManager] saveAll
[2023-06-22 11:57:46.794] [debug] [MainWindow] Can't send update-downloads-dropdown, will retry
[2023-06-22 11:57:46.794] [debug] [DownloadsManager] checkForDeletedFiles
[2023-06-22 11:57:46.813] [error] Logger Log level set to: info
[2023-06-22 11:57:46.821] [warn]  [App.Initialize] Current working directory is /, changing into /Applications/Mattermost.app/Contents/MacOS
[2023-06-22 11:57:46.915] [info]  [App.Initialize] Autoupgrade disabled: false
[2023-06-22 11:57:46.949] [info]  [ServerDropdownView] init
[2023-06-22 11:57:46.954] [info]  [i18nManager] Set new language fr
[2023-06-22 11:57:47.355] [error] [serverAPI] Error parsing server data from ***myserveur***/aows/api/v4/config/client?format=old
[2023-06-22 11:57:47.355] [warn]  [App.Utils] Could not get server info for AOWS Error: Error parsing server data from ***myserveur***/aows/api/v4/config/client?format=old
    at IncomingMessage.<anonymous> (/Applications/Mattermost.app/Contents/Resources/app.asar/index.js:75433:71)
    at IncomingMessage.emit (node:events:513:28)
    at endReadableNT (node:internal/streams/readable:1359:12)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21)
```

### Additional Information

I've tried to uninstall / and downgrade to 5.3.1 or 5.4-rc1 but same thing happens.
I'm able to join my team with the mattermost browser online.
I'm able to join another server with my app and switch channel/pm.

**Comment by asplamagnifique2** [comment]
<img width="1060" alt="Capture d’écran 2023-06-22 à 15 52 29" src="https://github.com/mattermost/desktop/assets/77117428/99270c7e-4edd-4f8d-9e69-7b304059f84c">

Sure.
Here what I can see if I want to switch to another channel.
The link return is not working, I've to Ctrl+SHIFT+R to refresh and go back to the main channel or quit and relaunch the app

**Comment by devinbinnie** [comment]
@asplamagnifique2 Does removing and re-adding your server fix the issue?

**Comment by asplamagnifique2** [comment]
oh you gave me an idea and it's working : first (this morning) when I've delete the server and recreate it, I had the same issue.

So I've just retried, but this time, I gave to it another name, as on the log error message, the were an information about the direcroty used to store my settings.

And it's now working fine, I can switch again between channel !

<img width="133" alt="Capture d’écran 2023-06-22 à 15 58 15" src="https://github.com/mattermost/desktop/assets/77117428/54b7578d-5437-4897-9546-49f1aaa6d2bc">

I think my issue was about the directory used to store some settings linked to the server name set on the app.
